benchmarks/src/main.rs

```diff
@@ -270,6 +272,95 @@ async fn parallel_benchmarks(uri: &str, more_info: bool, ids: &[bool]) -> Result
+            num_iter: 10_000,
+            path: DATA_PATH.join("extended_bson").join("flat_bson.json"),
+        };
+        println!("Running BSON flat decode...");
+        let bson_flat_decode =
+            bench::run_benchmark::<BsonDecodeBenchmark>(bson_flat_decode_options).await?;
+
+        comp_score += score_test(bson_flat_decode, "BSON flat decode", 75.31, more_info);
+    }
+
+    // BSON flat document encode
+    if ids[10] {
+        let bson_flat_encode_options = bench::bson_encode::Options {
+            num_iter: 10_000,
+            path: DATA_PATH.join("extended_bson").join("flat_bson.json"),
+        };
+        println!("Running BSON flat encode...");
+        let bson_flat_encode =
+            bench::run_benchmark::<BsonEncodeBenchmark>(bson_flat_encode_options).await?;
+
+        comp_score += score_test(bson_flat_encode, "BSON flat encode", 75.31, more_info);
+    }
+
+    // BSON deep document encode
+    if ids[11] {
+        let bson_deep_decode_options = bench::bson_decode::Options {
+            num_iter: 10_000,
+            path: DATA_PATH.join("extended_bson").join("deep_bson.json"),
+        };
+        println!("Running BSON deep decode...");
+        let bson_deep_decode =
+            bench::run_benchmark::<BsonDecodeBenchmark>(bson_deep_decode_options).await?;
+
+        comp_score += score_test(bson_deep_decode, "BSON deep decode", 19.64, more_info);
+    }
+
+    // BSON deep document encode
+    if ids[12] {
+        let bson_deep_encode_options = bench::bson_encode::Options {
+            num_iter: 10_000,
+            path: DATA_PATH.join("extended_bson").join("deep_bson.json"),
+        };
+        println!("Running BSON deep encode...");
+        let bson_deep_encode =
+            bench::run_benchmark::<BsonEncodeBenchmark>(bson_deep_encode_options).await?;
+
+        comp_score += score_test(bson_deep_encode, "BSON deep encode", 19.64, more_info);
+    }
+
+    // BSON full document encode
+    if ids[13] {
+        let bson_full_decode_options = bench::bson_decode::Options {
+            num_iter: 10_000,
+            path: DATA_PATH.join("extended_bson").join("full_bson.json"),
+        };
+        println!("Running BSON full decode...");
+        let bson_full_decode =
+            bench::run_benchmark::<BsonDecodeBenchmark>(bson_full_decode_options).await?;
+
+        comp_score += score_test(bson_full_decode, "BSON full decode", 57.34, more_info);
+    }
+
+    // BSON full document encode
+    if ids[14] {
+        let bson_full_encode_options = bench::bson_encode::Options {
+            num_iter: 10_000,
+            path: DATA_PATH.join("extended_bson").join("full_bson.json"),
+        };
+        println!("Running BSON full encode...");
+        let bson_full_encode =
+            bench::run_benchmark::<BsonEncodeBenchmark>(bson_full_encode_options).await?;
+
+        comp_score += score_test(bson_full_encode, "BSON full encode", 57.34, more_info);
```

Comment:
From looking at the implementation of `score_test` it appears to be a little broken in that it relies on `MAX_ITERATIONS`  when it doesn't need to. The only iteration count it should consider is the number of iterations that actually occurred. This is something that I should've caught last time, but fortunately there's a commit in this PR for last ticket anyways :).

Reply:
Updated to use `len() / 2` instead of `MAX_ITERATIONS / 2`.

Reply:
That wasn't quite what I was suggesting. I left a few more specific comments up there on the actual code now that there are diffs up there (:shakefist: at GitHub for not letting me comment outside of diffs).